From f080cb6f623270bb7fa658f0fa84c498a988aaef Mon Sep 17 00:00:00 2001
From: Thao Luong <thao.luong.uw@rvc.renesas.com>
Date: Wed, 27 May 2015 11:40:26 +0700
Subject: [PATCH 19/25] gstomx: Once reinitialize an instance and then retry 
 OMX_Init() if OMX_GetHandle() fails

Reinitializing an instance in case that OMX_GetHandle() fails may
be effective especially for some stateful OMXIL libraries since it
resets the state.

refs #69206
Signed-off-by: Thao Luong <thao.luong.uw@rvc.renesas.com>
---
 omx/gstomx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/omx/gstomx.c b/omx/gstomx.c
index 5a4c242..4d75fa2 100644
--- a/omx/gstomx.c
+++ b/omx/gstomx.c
@@ -669,7 +669,9 @@ gst_omx_component_new (GstObject * parent, const gchar * core_name,
   GstOMXCore *core;
   GstOMXComponent *comp;
   const gchar *dot;
+  gint retry = 1;
 
+reinit:
   core = gst_omx_core_acquire (core_name);
   if (!core)
     return NULL;
@@ -692,6 +694,8 @@ gst_omx_component_new (GstObject * parent, const gchar * core_name,
     gst_omx_core_release (core);
     g_free (comp->name);
     g_slice_free (GstOMXComponent, comp);
+    if (retry-- > 0)
+      goto reinit;
     return NULL;
   }
   GST_DEBUG_OBJECT (parent,
-- 
2.4.0

